pytorch_lightning/utilities/distributed.py

```diff
@@ -301,6 +301,21 @@ def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
 
 
+def get_default_process_group_backend_for_device(device: torch.device) -> str:
+    return "nccl" if device.type == "cuda" else "gloo"
+
+
+def _get_process_group_backend_from_env() -> Optional[str]:
```

Comment:
why this is a private method?

Reply:
Since this PR deprecates the env var support, I didn't want to expose a specific public API that would also be deprecated/removed later